docs: Prepare Changelog for 8.7.1 (#35115)
elastic/beats: CHANGELOG.asciidoc

:issue: https://github.com/elastic/beats/issues/
:pull: https://github.com/elastic/beats/pull/

[[release-notes-8.7.1]]
=== Beats version 8.7.1
https://github.com/elastic/beats/compare/v8.7.0\...v8.7.1[View commits]

==== Bugfixes

> *Affecting all Beats*
> - Fix Beats started by agent do not respect the allow_older_versions: true configuration flag {issue}34227[34227] {pull}34964[34964]
> - Fix performance issues when we have a lot of inputs starting and stopping by allowing to disable global processors under fleet. {issue}35000[35000] {pull}35031[35031]
> 
> *Filebeat*
> - [GCS Input] Added missing locks for safe concurrency {pull}34914[34914]
> - Add input instance id to request trace filename for httpjson and cel inputs {pull}35037[35037]
> - Fix panic in TCP and UDP inputs on Linux when collecting socket metrics from OS. {issue}35064[35064]
> 
> *Heartbeat*
> - Fix issue using projects in airgapped environments by disabling npm audit. {pull}34936[34936]
> 
> *Packetbeat*
> - Fix documentation for `flows.period` related to flow reporting. {pull}35009[35009]
> 
> ==== Added
> 
> *Affecting all Beats*
> - Allow users to enable features via configuration, starting with the FQDN reporting feature. {issue}1070[1070] {pull}34456[34456]
> 
> *Winlogbeat*
> - Add note in documentation about 21 event ID clause limit {issue}35048[35048] {pull}35049[35049]

Reviewer: ```suggestion
*Affecting all Beats*

- Fix Beats started by agent do not respect the allow_older_versions: true configuration flag {issue}34227[34227] {pull}34964[34964]
- Fix performance issues when we have a lot of inputs starting and stopping by allowing to disable global processors under fleet. {issue}35000[35000] {pull}35031[35031]

*Filebeat*

- [GCS Input] Added missing locks for safe concurrency {pull}34914[34914]
- Add input instance id to request trace filename for httpjson and cel inputs {pull}35037[35037]
- Fix panic in TCP and UDP inputs on Linux when collecting socket metrics from OS. {issue}35064[35064]

*Heartbeat*

- Fix issue using projects in airgapped environments by disabling npm audit. {pull}34936[34936]

*Packetbeat*

- Fix documentation for `flows.period` related to flow reporting. {pull}35009[35009]

==== Added

*Affecting all Beats*

- Allow users to enable features via configuration, starting with the FQDN reporting feature. {issue}1070[1070] {pull}34456[34456]

*Winlogbeat*

- Add note in documentation about 21 event ID clause limit {issue}35048[35048] {pull}35049[35049]
```
Looks like the script that generates these could use a small update. For a bulleted list in Markdown we need a blank line after the heading. 

![Screenshot 2023-04-18 at 12 13 51 PM](https://user-images.githubusercontent.com/41695641/232839552-d171cea1-45b2-4f71-b49f-5618671d3fc3.png)

Reviewer 2: No, that was my doing - to remove the blank lines after each heading.  I didn't realize they were significant.  I'll fix this now - thanks for checking and calling it out, @kilfoyle!

Reviewer 2: Fixed in f5ef618d6c67239487b7aed2779bb9d4fef393a0.  Sorry, I just noticed your suggested change in the PR 🤦.

:issue: https://github.com/elastic/beats/issues/
:pull: https://github.com/elastic/beats/pull/

[[release-notes-8.7.1]]
=== Beats version 8.7.1
https://github.com/elastic/beats/compare/v8.7.0\...v8.7.1[View commits]

==== Bugfixes

*Affecting all Beats*

- Fix Beats started by agent do not respect the allow_older_versions: true configuration flag {issue}34227[34227] {pull}34964[34964]

> - Fix performance issues when we have a lot of inputs starting and stopping by allowing to disable global processors under fleet. {issue}35000[35000] {pull}35031[35031]

Reviewer 3: ```suggestion
- Beats started by agent now configure one global instance of each default add_x_metadata processor instead of creating one processor per input. Fixes a significant performance degradation for the aws-s3 input. {issue}35000[35000] {pull}35031[35031]
```

This changelog entry isn't quite right, the primary fix was creating the default processors as Beat global processors.

:issue: https://github.com/elastic/beats/issues/
:pull: https://github.com/elastic/beats/pull/

[[release-notes-8.7.1]]
=== Beats version 8.7.1
https://github.com/elastic/beats/compare/v8.7.0\...v8.7.1[View commits]

==== Bugfixes

*Affecting all Beats*

> - Fix Beats started by agent not respecting the allow_older_versions: true configuration flag {issue}34227[34227] {pull}34964[34964]

Reviewer: ```suggestion
- Fix Beats started by agent not respecting the `allow_older_versions: true` configuration flag {issue}34227[34227] {pull}34964[34964]
```

> - Beats started by agent now configure one global instance of each default add_x_metadata processor instead of creating one processor per input. Fixes a significant performance degradation for the aws-s3 input. {issue}35000[35000] {pull}35031[35031]

Reviewer: ```suggestion
- Beats started by agent now configure one global instance of each default add_x_metadata processor instead of creating one processor per input. Fixes a significant performance degradation for the `aws-s3` input. {issue}35000[35000] {pull}35031[35031]
```

*Filebeat*

- [GCS Input] Added missing locks for safe concurrency {pull}34914[34914]

> - Add input instance id to request trace filename for httpjson and cel inputs {pull}35037[35037]

Reviewer: ```suggestion
- Add input instance ID to request trace filename for `httpjson` and `cel` inputs {pull}35037[35037]
```

- Fix panic in TCP and UDP inputs on Linux when collecting socket metrics from OS. {issue}35064[35064]
- Correctly collect TCP and UDP metrics for unspecified address values. {pull}35111[35111]

*Heartbeat*

> - Fix issue using projects in airgapped environments by disabling npm audit. {pull}34936[34936]

Reviewer: ```suggestion
- Fix issue using projects in air-gapped environments by disabling npm audit. {pull}34936[34936]
```
